### PR TITLE
Add support for systemd when bootstrapping machine executor

### PR DESCRIPTION
**Pull request by konstan**
- Added `systemd` service configuration files for orchestrator and node machine executor services.
- Added `systemd` based configuration of the services into `slipstream.bootstrap`.
- Updated the `initd`/`systemd` distribution versions matrix and detection of the `initd`/`systemd` based distributions.
- Improvements
   - catch errors to start executor and publish abort message
   - explicitly fail during configuration of startup scripts.

Connected to #213

### CODE REVIEW

File: client/src/main/scripts/slipstream.bootstrap.py [hunk 1]
```diff
@@ -381,29 +403,80 @@ def _add_executor_to_initd(executor_name):
...
+    paths = ['/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin']
+    paths.append(os.path.join(SLIPSTREAM_CLIENT_HOME, 'bin'))
+    paths.append(os.path.join(SLIPSTREAM_CLIENT_HOME, 'sbin'))
+    conf['PATH'] = os.pathsep.join(paths)
+    conf['PYTHONPATH'] = os.path.join(SLIPSTREAM_CLIENT_HOME, 'lib')
+    conf['DAEMON_ARGS'] = _get_verbosity()
+    conf['SLIPSTREAM_CONNECTOR_INSTANCE'] = os.environ.get('SLIPSTREAM_CONNECTOR_INSTANCE')
+    if executor_name == 'orchestrator':
+        cloud_name = os.environ['SLIPSTREAM_CLOUD']
+        pypath_prep = conf.get('PYTHONPATH', '') and (conf.get('PYTHONPATH') + os.pathsep) or ''
```
Comment (schaubl) [inline client/src/main/scripts/slipstream.bootstrap.py]: It's better to use 
```python
pypath_prep = conf.get('PYTHONPATH') + os.pathsep if conf.get('PYTHONPATH', '') else ''
```

So it prevent this kind of problems:
```python
>>> 'x' in 'abc' and 0 or 1
1 (OK)
>>> 'b' in 'abc' and 0 or 1
1 (NOT OK)

>>> 0 if 'x' in 'abc' else 1
1 (OK)
>>> 0 if 'b' in 'abc' else 1
0 (OK)
```

File: client/src/main/scripts/slipstream.bootstrap.py [hunk 2]
```diff
@@ -381,29 +403,80 @@ def _add_executor_to_initd(executor_name):
...
-        fh.write('export SLIPSTREAM_CONNECTOR_INSTANCE="%s"\n' % os.environ.get('SLIPSTREAM_CONNECTOR_INSTANCE'))
-        if executor_name == 'orchestrator':
-            cloud_name = os.environ['SLIPSTREAM_CLOUD']
-            fh.write('export PYTHONPATH=$PYTHONPATH:%s\n' % os.path.join(os.sep, 'opt', cloud_name.lower()))
-
-            env_matcher = re.compile('SLIPSTREAM_')
-            for var, val in os.environ.items():
-                if env_matcher.match(var) and var != 'SLIPSTREAM_NODE_INSTANCE_NAME':
-                    #if re.search(' ', val) and not (val.startswith('"') and val.endswith('"')):
-                    if re.search(' ', val) and not 'SLIPSTREAM_COOKIE':
```
Comment (schaubl) [inline client/src/main/scripts/slipstream.bootstrap.py]: @konstan: Do you know why we were filtering `SLIPSTREAM_COOKIE` and why the previous line has been commented ?

Comment (konstan) [inline client/src/main/scripts/slipstream.bootstrap.py]: I'm afraid this was committed in by mistake.  That `not 'SLIPSTREAM_COOKIE'` doesn't make sense at all.

File: client/src/main/scripts/etc/slipstream-orchestrator.service [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=SlipStream Orchestrator Executor
+After=syslog.target
```
Comment (schaubl) [inline client/src/main/scripts/etc/slipstream-orchestrator.service]: Maybe we can replace it with:
```ini
[Unit]
Description=SlipStream Orchestrator Executor
Wants=network-online.target
After=syslog.target network.target network-online.target
```

This comment also apply to the file `slipstream-node.service`

You can find some details about waiting on the network access here http://unix.stackexchange.com/questions/126009/cause-a-script-to-execute-after-networking-has-started

File: client/src/main/scripts/slipstream.bootstrap.py (hunk 1 above)
Comment (konstan) [inline client/src/main/scripts/slipstream.bootstrap.py]: Yes, I'm aware of that thing when parameter after `end` evaluates to `False` :)  But in this particular case of `(conf.get('PYTHONPATH') + os.pathsep)` my rational was that it never going to become `False`.  And this is my old (probably bad) habit to write that way.   Let me change here to what you are suggesting though.

**Pull request merged**